Example/Sources/ConversationViewController.swift

```diff
@@ -0,0 +1,106 @@
+                   "Quam erat vehicula metus, et condimentum ante tellus augue."
+        
+        let msg5 = "Lorem ipsum dolor sit amet, consectetur adipiscing elit." +
+                   "Pellentesque venenatis, ante et hendrerit rutrum" +
+                   "Quam erat vehicula metus, et condimentum ante tellus augue."
+
+        messages.append(MockMessage(text: msg2, sender: sender2, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg4, sender: currentSender(), id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg5, sender: sender3, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg1, sender: currentSender(), id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg3, sender: sender1, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg3, sender: sender1, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg2, sender: sender2, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg2, sender: sender2, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg1, sender: currentSender(), id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg3, sender: sender1, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg2, sender: sender2, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg4, sender: currentSender(), id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg5, sender: sender3, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg4, sender: currentSender(), id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg5, sender: sender3, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg4, sender: currentSender(), id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg5, sender: sender3, id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg1, sender: currentSender(), id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg1, sender: currentSender(), id: NSUUID().uuidString))
+        messages.append(MockMessage(text: msg3, sender: sender1, id: NSUUID().uuidString))
+    }
+    
+    func currentSender() -> Sender {
+        return Sender(id: "123", displayName: "Steven")
+    }
+    
+    func numberOfMessages(in collectionView: UICollectionView) -> Int {
+        return messages.count
+    }
+    
+    func messageForItem(at indexPath: IndexPath, in collectionView: UICollectionView) -> MessageType {
+        return messages[indexPath.section]
+    }
+    
+    func avatarForMessage(_ message: MessageType, at indexPath: IndexPath, in collectionView: UICollectionView) -> Avatar {
+        let image = isFromCurrentSender(message: message) ? #imageLiteral(resourceName: "Steve-Jobs") : #imageLiteral(resourceName: "Tim-Cook")
+        return Avatar(placeholderImage: image)
```

Comment:
Hey @lzyraining thanks for the feedback!

Deciding how to expose an API that allows for easy customization will be a difficult task. I hope to release a really basic v0.1.0 beta in the next few hours and will be looking for feedback/suggestions.

As for your suggestion, I don't think `Sender` is a good fit for the responsibility of Avatar because some people may not even want Avatars. I'd like to keep this types responsibility minimal. I've even considered making it a protocol.